File: misc/aooswd/ex/strategy/java/MemoryBlock.java
/*
 * MemoryBlock.java
 *
 * Created on January 23, 2003, 10:42 AM
 */

/**
 *
 * @author  dgayler
 */
public class MemoryBlock
{
    
    private int lower;
    
    private int upper;
    
    /** Creates a new instance of MemoryBlock */
    public MemoryBlock(int lower, int upper) throws AddressOrderException
    {
        if (upper < lower || lower < 0 || upper < 0)
            throw new AddressOrderException ("invalid block size");
        this.lower = lower;
        this.upper = upper;
    }
    
    public int size()
    {
        return upper - lower + 1;
    }
    
    /** Getter for property lower.
     * @return Value of property lower.
     */
    public int getLower()
    {
        return lower;
    }
    
    /** Getter for property upper.
     * @return Value of property upper.
     */
    public int getUpper()
    {
        return upper;
    }
    
    /** Setter for property lower.
     * @param lower New value of property lower.
     */
    public void setLower(int lower) throws AddressOrderException
    {
        if (lower > upper)
             throw new AddressOrderException ("invalid block size");       
        this.lower = lower;
    }
    
    /** Setter for property upper.
     * @param upper New value of property upper.
     */
    public void setUpper(int upper) throws AddressOrderException
    {
        if (upper < lower)
             throw new AddressOrderException ("invalid block size");       
        this.upper = upper;
    }
    
    public String toString()
    {
        return "[" + lower + "," + upper + "]";
    }
    
}

File: misc/aooswd/ex/strategy/cpp/memory_block.cpp
#include "memory_block.h"

MemoryBlock::MemoryBlock (int l, int u) throw (AddressOrderException)
{
  if (l < 0 || u < l)
    throw AddressOrderException ("invalid memory block");
  lower = l;
  upper = u;
}

void MemoryBlock::setLower (int l) throw (AddressOrderException)
{
  if (l > upper)
    throw AddressOrderException ("invalid memory block");
  lower = l;
}

void MemoryBlock::setUpper (int u) throw (AddressOrderException)
{
  if (u < lower)
    throw AddressOrderException ("invalid memory block");
  upper = u;
}

ostream& operator << (ostream& out, const MemoryBlock& b)
{
  out << '[' << b.getLower() << ',' << b.getUpper() << ']';
}

